42/README.md

```diff
@@ -70,14 +71,129 @@ The above process ensure the node keeps validating and accepting new blocks whil
-- The validator set at height `h` should be used as the validator set in the genesis of the new blockchain.
-- Since from Tendermint's point of view this is a new blockchain, block height will start from `0` which implies that BigchainDB would need to archive the existing blocks collection. All other collections would remain as is.
+
+This section advises on how to approach backwards-incompatible upgrades of the Tendermint chain - the type of updgrades when Tendermint can not build blocks on top of the existing chain.
+
+BigchainDB operators need a convenient way to do the following:
+
+1. Stop building blocks when the old Tendermint chain is at a jointly chosen height.
+2. Continue building blocks using the new Tendermint chain starting from that height.
+3. Replay the whole blockchain after a migration.
+4. Join the network as a new validator after a migration.
+
+Additionally, BigchainDB HTTP API usage has to be seamless - all the HTTP responses has to look like there were no migrations.
+
+Below we describe how to perform a migration while meeting the postulated requirements. Note that there is no limitation on the number of consequent migrations that can be performed this way.
+
+#### 1. Stop building blocks
+
+A CLI command can be used to instruct the node to stop building blocks at a chosen height.
+
+To make sure no blocks are committed after the agreed height, we propose an election process:
+
+- The initiator creates an election, which includes a block height.
+- The validators vote for the election. Once the election is concluded, new blocks are not committed by the validators. New transactions from the users are rejected. New blocks sent by Tendermint are rejected too. The conditions for resuming the chain operation are described below.
+
+To perform a migration election, we propose 2 CLI commands.
+
+The initiator executes:
+
+```
+$ bigchaindb migration new <height> --private-key /home/user/.tendermint/config/priv_validator.json
+```
+
+The command outputs the migration ID. The initiator distributes it among other members of the network. The process is similar to adding new validators.
+
+Validators vote for the migration:
+
+```
+$ bigchaindb migration approve <migration-id> --private-key /home/user/.tendermint/config/priv_validator.json
+```
+
+Validators can watch how the election goes:
+
+```
+$ bigchaindb migration status <migration-id>
+```
+
+Outputs:
+```
+votes_recieved=<Sum_of_votes_recieved>
+votes_allocated=<Sum_of_votes_allocated_in_election>
+network_size=<Total_network_power>
+```
+
+#### 2. Replay the blockchain
+
+In order to replay the chain up to the migration height, we need to either keep the old Tendermint version running or skip replaying the corresponding part of the chain = archive the chain. We consider the former to be a huge hassle so we conceive an archiving scheme further in this chapter.
```

Comment:
The last sentence ends ambiguously with, "...so we conceive an archiving scheme further in this chapter." Conceive? Chapter?

Maybe, "...so we describe an archiving scheme elsewhere in this BEP."